processor.go

```diff
@@ -504,7 +504,7 @@ func (p *processor) handleFailedMessage(ctx context.Context, msg *base.TaskMessa
 		p.retry(ctx, msg, err, false /*isFailure*/)
 		return
 	}
-	if msg.Retried >= msg.Retry || errors.Is(err, SkipRetry) {
+	if msg.Retried >= msg.Retry || errors.Is(err, SkipRetry) || errors.Is(err, context.Canceled) {
 		p.logger.Warnf("Retry exhausted for task id=%s", msg.ID)
```

Comment:
maybe - depending on case - change log message & level:

```suggestion
	if msg.Retried >= msg.Retry || errors.Is(err, SkipRetry) || errors.Is(err, context.Canceled) {
		if msg.Retried >= msg.Retry {
			p.logger.Warnf("Retry exhausted for task id=%s", msg.ID)
		} else {
			p.logger.Debugf("Archiving task id=%s (err:%v)", msg.ID, err)
		}
```